v2/internal/controllers/cognitiveservices_account_20250601_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	cognitiveservices "github.com/Azure/azure-service-operator/v2/api/cognitiveservices/v1api20250601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_CognitiveServices_Account_20250601_CRUD(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+
+	// Moving test to a different region:
+	// AI Foundry not available in westus2: https://learn.microsoft.com/en-us/azure/ai-foundry/reference/region-support
+	tc.AzureRegion = to.Ptr("westus3")
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	account := &cognitiveservices.Account{
+		ObjectMeta: tc.MakeObjectMetaWithName(tc.NoSpaceNamer.GenerateName("cogsvcacc")),
+		Spec: cognitiveservices.Account_Spec{
+			Identity: &cognitiveservices.Identity{
+				Type: to.Ptr(cognitiveservices.Identity_Type_SystemAssigned),
+			},
+			Kind:     to.Ptr("OpenAI"),
+			Location: tc.AzureRegion,
+			OperatorSpec: &cognitiveservices.AccountOperatorSpec{
+				Secrets: &cognitiveservices.AccountOperatorSecrets{
+					Key1:      &genruntime.SecretDestination{Name: "cogsecrets", Key: "key1"},
+					Key2:      &genruntime.SecretDestination{Name: "cogsecrets", Key: "key2"},
+					Endpoint:  &genruntime.SecretDestination{Name: "cogsecrets", Key: "endpoint"},
+					Endpoints: &genruntime.SecretDestination{Name: "cogsecrets", Key: "endpoints"},
+				},
+			},
+			Owner: testcommon.AsOwner(rg),
+			Properties: &cognitiveservices.AccountProperties{
+				PublicNetworkAccess: to.Ptr(cognitiveservices.AccountProperties_PublicNetworkAccess_Enabled),
+			},
+			Sku: &cognitiveservices.Sku{
+				Name: to.Ptr("S0"),
+			},
+		},
+	}
+
+	tc.CreateResourcesAndWait(account)
+	tc.Expect(account.Status.Id).ToNot(BeNil())
+	tc.ExpectSecretHasKeys("cogsecrets", "key1", "key2", "endpoint", "endpoints")
+
+	oldAcc := account.DeepCopy()
+	if oldAcc.Spec.Tags == nil {
+		oldAcc.Spec.Tags = map[string]string{}
+	}
+	account.Spec.Tags = map[string]string{"env": "test"}
+	tc.PatchResourceAndWait(oldAcc, account)
+	tc.Expect(account.Status.Tags).To(Equal(map[string]string{"env": "test"}))
+
+	tc.RunParallelSubtests(
+		testcommon.Subtest{
+			Name: "CognitiveServices Deployment CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
```

Comment:
minor: Can you put this subtest into its own named function?

You still call it here just do: `Test: funcName` rather than the whole function def.